### PR TITLE
fix(color): simplify the current color depth macro definition

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
### Description of the feature or fix

by ## concat the macro 

### Checkpoints
- [X] Follow the [styling guide](https://github.com/lvgl/lvgl/blob/master/docs/CODING_STYLE.md)
- [ ] Update CHANGELOG.md
- [ ] Update the documentation

### CODE REVIEW

File: src/lv_misc/lv_types.h [hunk 1]
```diff
@@ -74,6 +74,12 @@ typedef uint32_t lv_uintptr_t;
 
 #define LV_UNUSED(x) ((void) x)
 
+#define LV_CONCAT_(x, y) x ## y
```
Comment (kisvegabor) [inline src/lv_misc/lv_types.h]: Please change it to `_LV_CONCAT`.
We denote internal API names with `_` prefix.

**Review by kisvegabor — COMMENTED** [review]
I like the changes! I had only a minor comemnt.

File: src/lv_misc/lv_types.h (hunk 1 above)
Comment (xiaoxiang781216) [reply src/lv_misc/lv_types.h]: Done.

Comment (kisvegabor) [reply src/lv_misc/lv_types.h]: Thanks!

**Comment by kisvegabor** [comment]
Looks good to me, let's wait for @embeddedt's review.

**Comment by embeddedt** [comment]
@xiaoxiang781216 This is the warning I get in a C file. C++ seems to be compiling without errors.

I wouldn't be so picky about `-Wpedantic`, except for the fact that it's a good indicator of where other compilers might fail.

```
lv_misc/lv_color.h:197:38: warning: initializer element is not constant [-Wpedantic]
  197 | # define LV_COLOR_MAKE32(r8, g8, b8) (_LV_COLOR_MAKE_TYPE_HELPER{{b8, g8, r8, 0xff}}) /*Fix 0xff alpha*/
      |                                      ^
lv_misc/lv_types.h:77:26: note: in expansion of macro 'LV_COLOR_MAKE32'
   77 | #define _LV_CONCAT(x, y) x ## y
      |                          ^
lv_misc/lv_types.h:78:25: note: in expansion of macro '_LV_CONCAT'
   78 | #define LV_CONCAT(x, y) _LV_CONCAT(x, y)
      |                         ^~~~~~~~~~
lv_misc/lv_color.h:213:35: note: in expansion of macro 'LV_CONCAT'
  213 | #define LV_COLOR_MAKE(r8, g8, b8) LV_CONCAT(LV_COLOR_MAKE, LV_COLOR_DEPTH)(r8, g8, b8)
      |                                   ^~~~~~~~~
lv_misc/lv_color.h:214:36: note: in expansion of macro 'LV_COLOR_MAKE'
  214 | #define _LV_COLOR_ZERO_INITIALIZER LV_COLOR_MAKE(0, 0, 0)
      |                                    ^~~~~~~~~~~~~
lv_misc/lv_color.h:536:43: note: in expansion of macro '_LV_COLOR_ZERO_INITIALIZER'
  536 |         static lv_color_t bg_color_save = _LV_COLOR_ZERO_INITIALIZER;
      |                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by xiaoxiang781216** [comment]
Fixed, it's the same warning happen: https://github.com/lvgl/lvgl/pull/2072.

**Review by embeddedt — APPROVED** [review]

**Comment by embeddedt** [comment]
Perfect; thank you!

**Comment by xiaoxiang781216** [comment]
BTW, can we add -Wpedantic to the CI? So the similar warning can be catched automatically.

**Comment by embeddedt** [comment]
Yes; will do.

**Comment by guinn8** [comment]
In case anyone else is getting `error: unknown type name 'lv_color_t'; did you mean 'lv_color8_t'?` when upgrading from v7.9 this commit is to blame

**Comment by xiaoxiang781216** [comment]
@guinn8 could you give the more detail information you hit?

**Comment by guinn8** [comment]
@xiaoxiang781216 
[trace.txt](https://github.com/lvgl/lvgl/files/6352455/trace.txt)
Compiler: gcc-arm-none-eabi-9-2020-q2-update
SDK/Board: SDK_2.9.1_EVK-MIMXRT1020

**Comment by embeddedt** [comment]
I could be wrong, but the issue might be caused by the fact that `LV_COLOR_DEPTH` is not defined as a constant. Instead there is an additional layer of macro indirection, which preprocessors tend to choke on.

**Comment by xiaoxiang781216** [comment]
Yes, what's your LV_COLOR_DEPTH definition?

**Pull request merged**